ldap/servers/slapd/back-ldbm/db-bdb/bdb_layer.c

```diff
@@ -4010,9 +4074,13 @@ bdb_checkpoint_threadmain(void *param)
             compaction_scheduled = PR_TRUE;
             slapi_eq_once_rel(bdb_compact, (void *)li, slapi_current_rel_time_t() + compactdb_time);
 
-            /* reset interval timer */
-            compactdb_interval = compactdb_interval_update;
-            slapi_timespec_expire_at(compactdb_interval, &compactdb_expire);
+            /* reset compact interval timer */
+            compactdb_interval_orig = compactdb_interval_update;
+            slapi_timespec_expire_at(compactdb_interval_update, &compactdb_expire);
+            /* lastly update the config */
+            compactdb_start_time = slapi_current_utc_time();
+            slapi_eq_once_rel(bdb_write_compact_start_time, (void *)li,
+                              slapi_current_rel_time_t() + 3);
```

Comment:
I'm not very proficient with BDB, but this 3-second delay seems a bit strange to me.
Why exactly 3 seconds? And if it's a precautioon, can we do it any other way? (like check when the BDB is ready or something with some 100ms retries?)

Reply:
> I'm not very proficient with BDB, but this 3-second delay seems a bit strange to me. Why exactly 3 seconds? And if it's a precautioon, can we do it any other way? (like check when the BDB is ready or something with some 100ms retries?)

I'm not sure if there is an easy way to tell if all the plugins have been started from inside the check pointing thread.  The race condition I am avoiding usually all happens within the same second.  But the mod update doesn't need to happen right away, so I figured waiting 3 seconds was a safe bet